test-tools/proof-explorer/src/Root.tsx

```diff
@@ -0,0 +1,1006 @@
+            builder.addMembership(
+                attribute,
+                set.split(',').map((e) => e.trim())
+            );
+        } else {
+            builder.addNonMembership(
+                attribute,
+                set.split(',').map((e) => e.trim())
+            );
+        }
+
+        setStatement(builder.getStatement());
+    };
+
+    return (
+        <form>
+            <div className="form-group border rounded border-primary my-2 p-1">
+                <label>{`Prove ${attribute}${member ? ' ' : ' not '}in`} </label> <br />
+                <input className="my-1" onChange={onSetChange} value={set} />
+                <button onClick={onClickAdd} type="button" className="btn btn-primary">
+                    {'Add'}
+                </button>
+            </div>
+        </form>
+    );
+}
+
+interface SpecialSetProps extends ExtendStatementProps {
+    // if nationality is set then produce statement about EU nationality
+    // otherwise about EU residence
+    nationality: boolean;
+}
+
+function EUAttributeIn({ nationality, setStatement }: SpecialSetProps) {
+    const onClickAdd: MouseEventHandler<HTMLButtonElement> = () => {
+        const builder = new AccountStatementBuild();
+        if (nationality) {
+            builder.addEUNationality();
+        } else {
+            builder.addEUResidency();
+        }
+
+        setStatement(builder.getStatement());
+    };
+
+    return (
+        <form>
+            <div className="form-group border rounded border-primary my-2 p-1">
+                <label>{`Prove ${nationality ? 'nationality in EU' : 'residence in EU'}`} </label> <br />
+                <button onClick={onClickAdd} type="button" className="btn btn-primary">
+                    {'Add'}
+                </button>
+            </div>
+        </form>
+    );
+}
+
+function IdentityProviders({ idps }: { idps: { name: string; id: number }[] }): [number[], React.JSX.Element[]] {
+    const [checked, setChecked] = useState<number[]>([]);
+    const handleCheck = (event: ChangeEvent<HTMLInputElement>) => {
+        let updatedList = [...checked];
+        if (event.target.checked) {
+            updatedList = [...checked, parseInt(event.target.value)];
+        } else {
+            updatedList.splice(checked.indexOf(parseInt(event.target.value)), 1);
+        }
+        setChecked(updatedList);
+    };
+
+    return [
+        checked,
+        idps.map(({ name, id }) => (
+            <div className="form-check">
+                <input
+                    className="form-check-input"
+                    type="checkbox"
+                    value={id}
+                    id="flexCheckChecked"
+                    onChange={handleCheck}
+                    checked={checked.includes(id)}
+                />
+                <label className="form-check-label">{name}</label>
+            </div>
+        )),
+    ];
+}
+
+const REGISTRY_CONTRACT_REGISTRY_METADATA_RETURN_VALUE_SCHEMA =
```

Comment:
Add to a `constant` file to make it easier visible in case we need to update it.

Reply:
Moved there.